frontend/app/preview/dashboard/page.tsx

```diff
@@ -0,0 +1,251 @@
+			onOpen();
+		}, 10000);
+
+		return () => {
+			clearTimeout(dataTimeout);
+			clearTimeout(openTimeout);
+		};
+	}, []);
+
+	// Sort data based on selected key
+	useEffect(() => {
+		const sortData = () => {
+			const sorted = [...data];
+			const sortKey = Array.from(selectedKeys)[0];
+
+			switch (sortKey) {
+				case "Date (Newest)":
+					sorted.sort((a, b) => new Date(b.received_at).getTime() - new Date(a.received_at).getTime());
+					break;
+				case "Date (Oldest)":
+					sorted.sort((a, b) => new Date(a.received_at).getTime() - new Date(b.received_at).getTime());
+					break;
+				case "Company":
+					sorted.sort((a, b) => a.company_name.localeCompare(b.company_name));
+					break;
+				case "Job Title":
+					sorted.sort((a, b) => a.job_title.localeCompare(b.job_title));
+					break;
+				case "Status":
+					sorted.sort((a, b) => a.application_status.localeCompare(b.application_status));
+					break;
+				default:
+					break;
+			}
+			setSortedData(sorted);
+		};
+
+		if (data.length > 0) {
+			sortData();
+		}
+	}, [selectedKeys, data]);
+
+	// Handle sorting selection change and store it in localStorage
+	const handleSortChange = (keys: Set<string>) => {
+		const sortKey = Array.from(keys)[0];
+		localStorage.setItem("sortKey", sortKey);
+		setSelectedKeys(new Set([sortKey]));
+	};
+
+	// Handle CSV download
+	async function downloadCsv() {
+		setDownloading(true);
+		// Mock CSV generation (no api call)
+		const mockCsvContent =
+			"Company,Status,Received,Job Title,Subject,Sender\n" +
+			mockData
+				.map(
+					(item) =>
+						`${item.company_name},${item.application_status},${new Date(item.received_at).toLocaleDateString()},${item.job_title},${item.subject},${item.email_from}`
+				)
+				.join("\n");
+
+		const blob = new Blob([mockCsvContent], { type: "text/csv" });
+		const link = document.createElement("a");
+		const url = URL.createObjectURL(blob);
+		link.href = url;
+		link.download = `job_applications_${new Date().toISOString().split("T")[0]}.csv`;
+		document.body.appendChild(link);
+		link.click();
+		document.body.removeChild(link);
+		URL.revokeObjectURL(url);
+		setDownloading(false);
+	}
+
+	return (
+		<div className="p-6">
+			<Modal backdrop="blur" isOpen={isOpen} size="xl" onClose={onClose}>
+				<ModalContent>
+					{(onClose) => (
+						<>
+							<ModalHeader className="flex flex-col gap-1">
+								Enjoying the preview? Join the waitlist!
+							</ModalHeader>
+							<ModalBody>
+								<p>
+									Be the first to experience all the amazing features of our app! We're launching
+									soon, and you can get early access by joining our exclusive waitlist.
+								</p>
+								<p>
+									By joining the waitlist, you'll receive updates and an early invitation to unlock
+									the full potential of our platform.
+								</p>
+								<p>
+									Don't miss out — secure your spot today and be among the first to enjoy everything
+									we have to offer!
```

Comment:
```suggestion
By joining the waitlist, you'll receive updates on new features and an invitation to signup when we launch outside of beta.
```
Sounds less sales-y lol